django/template/defaulttags.py

```diff
@@ -1188,21 +1191,44 @@ def querystring(context, query_dict=None, **kwargs):
 
         {% querystring page=page_obj.next_page_number %}
 
-    A custom ``QueryDict`` can also be used::
+    A custom `QueryDict` can be used::
 
         {% querystring my_query_dict foo=3 %}
+
+    Or a `dict` can be used::
+
+        {% querystring my_dict foo=3 %}
+
+    As can a mix of multiple args and kwargs::
+
+        {% querystring my_dict my_query_dict foo=3 bar=None %}
     """
-    if query_dict is None:
-        query_dict = context.request.GET
-    query_dict = query_dict.copy()
-    for key, value in kwargs.items():
-        if value is None:
-            if key in query_dict:
-                del query_dict[key]
-        elif isinstance(value, Iterable) and not isinstance(value, str):
-            query_dict.setlist(key, value)
-        else:
-            query_dict[key] = value
+    try:
+        query_dict = context.request.GET.copy()
+    except AttributeError:
+        if not args:
+            raise
+        query_dict = QueryDict(mutable=True)
```

Comment:
I think this logic is changing the documented behavior. The docs says:

> This tag requires a QueryDict or dict instance, which defaults to request.GET if none is provided.

The docs also say:

> You can provide custom QueryDict or dict instances as positional arguments to be used instead of request.GET.

The way this code is written, it seems that `request.GET` is used as a starting point, if available, and then any argument is added on top of that. I think that is not correct, and we'd need:

* a test ensuring the documented behavior holds, and
* a change in this part to ensure `request.GET` is used only if not `args`.